.github/workflows/precommithooks.yml

```diff
@@ -0,0 +1,22 @@
+name: pre-commit-hooks
+
+on: [pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
```

Comment:
@slayoo we need to add a `PERSONAL_ACCESS_TOKEN` repository secret so that this can auto-commit formatting changes made by the action.

Reply:
will it "just work" after merging or do we need to grant some permission? Thanks @jb-mackay for all this work!

Reply:
It will "just work" as you have already added this token to the main repo :)